parsec/mca/device/device_gpu.c

```diff
@@ -1177,6 +1177,7 @@ parsec_default_gpu_stage_in(parsec_gpu_task_t        *gtask,
     for(int i = 0; i < task->task_class->nb_flows; i++) {
         if( !(flow_mask & (1U << i)) ) continue;
         source = gtask->sources[i];
+        assert(source->device_private != NULL);
```

Comment:
We cannot stage in data that does not exist. If we get here (because of RW or RO) the application requested data to be pushed onto the device. If the source has no memory then we cannot make up data. I consider this an error.